Build failure: mwprocapture

### Steps To Reproduce

Version V1.3.0.4326 of mwprocapture (the Linux driver for the Magewell Pro Capture family) FTBFS when building against Linux 6.4 or newer.

Steps to reproduce the behavior:

- Add `hardware.mwProCapture.enable = true;` to your Nix configuration.
- Add `boot.kernelPackages = pkgs.linuxPackages_latest;` to your Nix configuration.
- `sudo nixos-rebuild switch`

### Build log

```
error: builder for '/nix/store/dz5zsn28avray5b5xzz6zigjxk6zymnr-mwprocapture-1.3.0.4328-6.4.6.drv' failed with exit code 2;
       last 10 log lines:
       > /build/ProCaptureForLinux_4328/src/sources/avstream/mw-event-dev.c:158:21: error: too many arguments to function 'class_create'
       >   158 |     g_dev.c_class = class_create(THIS_MODULE, MW_EVENT_NAME);
       >       |                     ^~~~~~~~~~~~
       > /nix/store/p48papw46f5dvmrfh1yky8m3zm4p2k3f-linux-6.4.6-dev/lib/modules/6.4.6/source/include/linux/device/class.h:230:29: note: declared here
       >   230 | struct class * __must_check class_create(const char *name);
       >       |                             ^~~~~~~~~~~~
       > cc1: some warnings being treated as errors
       > make[2]: *** [/nix/store/p48papw46f5dvmrfh1yky8m3zm4p2k3f-linux-6.4.6-dev/lib/modules/6.4.6/source/scripts/Makefile.build:252: /build/ProCaptureForLinux_4328/src/sources/avstream/mw-event-dev.o] Error 1
       > make[1]: *** [/nix/store/p48papw46f5dvmrfh1yky8m3zm4p2k3f-linux-6.4.6-dev/lib/modules/6.4.6/source/Makefile:2032: /build/ProCaptureForLinux_4328/src] Error 2
       > make: *** [Makefile:33: all] Error 2
       For full logs, run 'nix log /nix/store/dz5zsn28avray5b5xzz6zigjxk6zymnr-mwprocapture-1.3.0.4328-6.4.6.drv'.
error: 1 dependencies of derivation '/nix/store/fx65lcgx8ylywg7jcjfm0p3vk337rqbk-etc-nixos.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/minsn9ik71p6j02jr4lq6bhh079dw64r-kernel-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mc0zmqdgi9rf7067j4vv4rv30snyxj0l-linux-6.4.6-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1vdpykpgng9pm0yrx7rcdvx75g253hjp-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i0w7f294m0blxnw7rai4xy6hb4wpg9qc-mwprocapture-1.3.0.4328-6.4.6_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3819ql0d5pcf0rm3gz7i728b06ia481z-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/80i0diwk5dp6yv7qc6scr737dk89hy3s-nixos-system-ripper-23.05.20230729.d57e8c5.drv' failed to build
```

### Additional context

I emailed Magewell support and received a driver update to address this issue. I am working on a pull request for nixpkgs.

### Notify maintainers

@MP2E 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.3.13, NixOS, 23.05 (Stoat), 23.05.20230729.d57e8c5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - nixpkgs: `/nix/store/qxgp86sdhxwzj0k89kz2aaw0xw7d8f6i-source`
```

--- a/pkgs/os-specific/linux/mwprocapture/default.nix
+++ b/pkgs/os-specific/linux/mwprocapture/default.nix
@@ -12,12 +12,12 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "mwprocapture";
-  subVersion = "4328";
+  subVersion = "4373";
   version = "1.3.0.${subVersion}-${kernel.version}";
 
   src = fetchurl {
     url = "https://www.magewell.com/files/drivers/ProCaptureForLinux_${subVersion}.tar.gz";
-    sha256 = "197l86ad52ijmmq5an6891gd1chhkxqiagamcchirrky4c50qs36";
+    sha256 = "sha256-/6q+6CTlgkHOgq1PF8dSPfl/xm/UFczr/AGkac2mXZ8=";
   };
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
@@ -60,7 +60,7 @@ stdenv.mkDerivation rec {
     homepage = "https://www.magewell.com/";
     description = "Linux driver for the Magewell Pro Capture family";
     license = licenses.unfreeRedistributable;
-    maintainers = with maintainers; [ MP2E ];
+    maintainers = with maintainers; [ flexiondotorg MP2E ];
     platforms = platforms.linux;
   };
 }

